torchaudio_contrib/augmentation.py

```diff
@@ -0,0 +1,151 @@
+import math
+import torch
+
+import torch.nn as nn
+from .layers import _ModuleNoStateBuffers, STFT
+from .functional import phase_vocoder
+
+
+class StretchSpecTime(_ModuleNoStateBuffers):
+    """
+    Stretch stft in time without modifying pitch for a given rate.
+
+    Args:
+
+        rate (float): rate to speed up or slow down by. Defaults to 1.
+        hop_len (int): Number audio of frames between STFT columns.
+            Defaults to 512.
+        num_bins (int, optional): number of filter banks from stft.
+            Defaults to 1025.
+    """
+
+    def __init__(self, rate=1., hop_len=512, num_bins=1025):
+
+        super(StretchSpecTime, self).__init__()
+
+        self.rate = rate
+        phi_advance = torch.linspace(
+            0, math.pi * hop_len, num_bins)[..., None]
+
+        self.register_buffer('phi_advance', phi_advance)
+
+    def forward(self, spect, rate=None):
+        if rate is None:
+            rate = self.rate
+        return phase_vocoder(spect, rate, self.phi_advance)
+
+    def __repr__(self):
+        param_str = '(rate={})'.format(self.rate)
+        return self.__class__.__name__ + param_str
+
+
+class GaussianNoise(_ModuleNoStateBuffers):
+    """
+    Apply gaussian noise to a spectrogram.
+
+    Args:
+        scale (float or tensor): standard deviation of the distribution.
+    """
+
+    def __init__(self, scale=1.):
+        super(GaussianNoise, self).__init__()
+        self.scale = torch.as_tensor(scale)
+        self.loc = torch.tensor(0.)
+
+    def forward(self, spect, scale=None):
```

Comment:
`spect` should be `mag_specgrams` or something like that?

Reply:
Yeah definitley, I was using `spect` as a placeholder until we have some consensus on the whole `mag_` / `real_` / `_gram` / `specgrams` debate! Once we do we'll have to refactor `layers` and `functional` too.

Reply:
Which reminds me per #47 to change the file structure soon.

Reply:
👍